api/src/modules/impact/impact.service.ts

```diff
@@ -96,9 +96,19 @@ export class ImpactService {
           entitiesWithPagination.entities,
           fetchSpecification,
         );
-        impactTableDto.materialIds = this.getIdsFromTree(
+
+        /*
+         * Filter out ids to only include descendants of
+         * paginated root elements
+         */
+        const materialIds: string[] = this.getIdsFromTree(
           entitiesWithPagination.entities,
         );
+        impactTableDto.materialIds = impactTableDto.materialIds
+          ? impactTableDto.materialIds.filter((value: string) =>
+              materialIds.includes(value),
+            )
+          : materialIds;
```

Comment:
Considering this code seems to be a bit lower level to what the function getImpactTable does, it might be worth it refactoring this piece of code into the getIdsFromTree function (since it is a private function and only used three times for the same purpose)
It would receive the impactTableDto.materialsIds in this case, as an optional arugment, and do the filtering the same way inside the getIdsFromTree function